### PR TITLE
add models directory

### PR DESCRIPTION
**Pull request by peacebonez**

### CODE REVIEW

File: models/Conversation.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    {
+      user: {
+        type: Schema.Types.ObjectId,
+        ref: "users", //references the users model
+      },
+    },
+  ],
+  messages: [
+    {
+      messageBody: {
```
Comment (ThinkTink) [inline models/Conversation.js:16]: At some point we'll probably to know what language the message is in. We could store that, or we could just query what the default language for the fromUser is

File: models/Invitation.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    ref: "users", //references the users model
+  },
+  toEmail: {
+    type: String,
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now,
+  },
+});
```
Comment (ThinkTink) [inline models/Invitation.js:21]: We'll also need a way to know which invitation is linked to a given referral url. We could add a field for this, or perhaps just use the _id

File: models/User.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  password: {
+    type: String,
+    required: true,
+  },
+  primaryLanguage: {
+    type: String,
+    required: true,
+  },
+  contacts: [this], // Self-referencing UserSchema.  not sure if this will work.
+  date: {
```
Comment (ThinkTink) [inline models/User.js]: Maybe a name like dateJoined would be more clear?

File: models/User.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
+const mongoose = require("mongoose");
+
+const Schema = mongoose.Schema;
+
+//create key for user avatar?
```
Comment (ThinkTink) [inline models/User.js]: We can makes this a `// TODO:` to come back to later

**Review by ThinkTink — COMMENTED** [review]
Looking good. Have you tested it out to see if `[this]` works on the user schema?

File: models/Conversation.js (hunk 1 above)
Comment (peacebonez) [reply models/Conversation.js:16]: How does something like adding an originalLanguage field look?
`messageBody: {
        fromUser: {
          type: Schema.Types.ObjectId,
          ref: "users", //references the users model
        },
        originalLanguage: {
          type: String,
        },
        text: {
          type: String,
        },
        createdAt: {
          type: Date,
          default: Date.now,
        },
      },`

File: models/Invitation.js (hunk 2 above)
Comment (peacebonez) [reply models/Invitation.js:21]: Was planning on using the unique _id field generated!

File: models/User.js (hunk 3 above)
Comment (peacebonez) [reply models/User.js]: Sounds good, making the change now

File: models/User.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+  primaryLanguage: {
+    type: String,
+    required: true,
+  },
+  contacts: [],
```
Comment (ThinkTink) [inline models/User.js]: I think we do need to define something here. We can probably use the way shared in this article: https://www.initialapps.com/mongoose-why-you-may-be-having-issues-populating-across-multiple-levels/

**Review by ThinkTink — APPROVED** [review]

File: models/User.js (hunk 5 above)
Comment (ThinkTink) [reply models/User.js]: ^ Approved to merge after you fix this one thing

Comment (peacebonez) [reply models/User.js]: fixing now.  Thank you

**Review by evanyang1 — APPROVED** [review]

**Pull request merged**